core/arch/interp.c

```diff
@@ -5975,6 +6151,88 @@ mangle_indirect_branch_in_trace(dcontext_t *dcontext, instrlist_t *trace,
         }
     }
 #    endif
+#elif defined(AARCH64)
+    instr_t *instr;
+    reg_id_t jump_target_reg;
+    reg_id_t scratch;
+    bool has_str_ldr;
+    /* Mangle jump to indirect branch lookup routine,
+     * based on indirect target not being equal to next block in trace.
+     * Original ibl lookup:
+     *      str tgt_reg, TLS_REG2_SLOT
+     *      mov tgt_reg, jump_target
+     *      b ibl_routine
+     * Now we rewrite it into:
+     *      str x0, TLS_REG0_SLOT
+     *      mov x0, #trace_next_target
+     *      eor x0, x0, jump_target
+     *      cbnz x0, trace_exit (ibl routine)
+     *      ldr x0, TLS_REG0_SLOT
+     */
+
+    /* Check and remove previous two patched instructions. */
+    has_str_ldr = false;
+    jump_target_reg = check_and_remove_patched_ibl(trace, targeter);
```

Comment:
Maybe rename `check_and_remove_patched_ibl` and `check_patched_ibl` to describe what they do.

Reply:
done